### DISCUSSION TITLE
supabase.auth.getUser returns null after google oauth signin

### DISCUSSION THREAD

**Discussion by KyleHu14** [supabase/supabase#21871] [Questions]
I am trying to implement basic oAuth login with supabase and am running into the issue of getUser returning null after going through the google oAuth flow. 

The code in question can be found in this repo : [Link](https://github.com/KyleHu14/job-tracker).

I followed the nextjs server auth tutorial. All I changed from the code that was given was the login function in ./client/src/app/login/actions.tsx where I used signInWithOAuth rather than the signin function. After going through the oauth flow, I get redirected to localhost:3000/dashboard but user is null. 

Is there something I'm missing or doing wrong? Any help would be appreciated and ty in advance.

I have middleware initialized, I am creating a server client in ./src/utils/supabase/server.ts.

**Comment by AryanK1511** [comment]
Hey @KyleHu14 I was able to fix this with Remix. Are you still looking for an answer for this?

**Reply by KyleHu14** [reply]
Hello, thank youfor reaching out and I apologize for replying so late! I ended up giving up on trying to get server auth to work and scraped the project. Feel free to share your solution here though!
